### PR TITLE
Add the ability to parse PEM files with certificates

### PR DESCRIPTION
**Pull request by bifurcation**

### CODE REVIEW

**Review by suhasHere — APPROVED** [review]

File: lib/hpke/include/hpke/certificate.h [hunk 1]
```diff
@@ -30,10 +30,13 @@ struct Certificate
   using ParsedName = std::map<int, std::string>;
 
   explicit Certificate(const bytes& der);
+  Certificate(std::unique_ptr<ParsedCertificate>&& parsed_cert_in);
```
Comment (dimarusyy) [inline lib/hpke/include/hpke/certificate.h:33]: explicit ?

File: lib/hpke/include/hpke/certificate.h [hunk 2]
```diff
@@ -53,4 +56,6 @@ struct Certificate
   const bytes raw;
 };
 
+bool operator==(const Certificate& lhs, const Certificate& rhs);
```
Comment (dimarusyy) [inline lib/hpke/include/hpke/certificate.h:59]: const ?

Comment (bifurcation) [reply lib/hpke/include/hpke/certificate.h:59]: You mean `const bool`?  Seems unnecessary.

Comment (dimarusyy) [reply lib/hpke/include/hpke/certificate.h:59]: Mainly `bool operatoor==(..) const;`

**Review by dimarusyy — APPROVED** [review]

**Pull request merged**